### ISSUE TITLE
PieChart not showing labels

### ISSUE THREAD

**Issue by DMarinsa** [valor-software/ng2-charts#1026]
**Reproduction of the problem**
I'm trying to get a Pie Chart made with the keys and values of an object.

component.ts its

```
import { Component, OnInit } from '@angular/core';
import { ProjectService } from '../../services/project.service';
import { ActivatedRoute } from '@angular/router';

@Component({
  selector: 'app-project-page',
  templateUrl: './project-page.component.html',
  styleUrls: ['./project-page.component.scss']
})
export class ProjectPageComponent implements OnInit {

  project: object = {};
  id: any;
  chartKeys: Array<string> = [];
  chartValues: Array<number> = [];

  constructor(private projectService: ProjectService, private route: ActivatedRoute) { }

  ngOnInit() {
    this.route.params.subscribe(params => {
      this.id = params['id'];
    });
    this.projectService.getById(this.id).subscribe(project => {
      console.log(project);
      this.project = project;
    });
    this.projectService.getProfile(this.id).subscribe((profile: any) => {
      this.chartKeys = Object.keys(profile.response);
      console.log(this.chartKeys);
      this.chartValues = Object.values(profile.response);
    });
  }
}
```

and my html is 

```
<div class="container">
  <h2>{{project.name}}</h2>
  <div style="display: block">
    <canvas baseChart
                [data]="chartValues"
                [labels]="chartKeys"
                [chartType]="'doughnut'"></canvas>
  </div>
</div>
```

This is what is showing

https://imgur.com/SXX3y98

**Comment by mariusl-a** [comment]
I noticed you added a fix, but have you tried using Observables?

I do this for all data subscriptions, doesn't matter if its a graph, table, information, etc. This way I can change the data in a different component, or even refresh data every 60 seconds and angular takes care of refreshing the views automatically. 👍 

**TS**
```
data$: Observable<any> = null;
  private _data: BehaviorSubject<any>;
  private dataStore: {
    data: any
  }
```
```
constructor(private dataService: DataService,
    private router: Router) {
    this.dataStore = { data: null };
    this._data = <BehaviorSubject<any>>new BehaviorSubject(null);
    this.data$ = this._data.asObservable();
  }
```
```

load() {
    this.dataService.load(AppSettings.ApiPath + '/' + this.Id).subscribe(data => {
      this.dataStore.data = data;
      this._data.next(Object.assign({}, this.dataStore).data);
    });
  }
```
```
ngOnInit() {
    this.load();
  }
```

**HTML**
```
<ng-container *ngIf="data$ | async; let graph">

    <div style="display: block">
        <canvas baseChart [datasets]="[{ data:graph.Data, backgroundColor: colors, datalabels: { align: 'center', color: '#ffffff', anchor: 'center'}}]"
            [labels]="graph.Labels" [options]="barChartOptions" [colors]="colors" [chartType]="barChartType"
            (chartHover)="chartHovered($event)" (chartClick)="chartClicked($event)"></canvas>
    </div>
</ng-container>
```

**Comment by 93Alliance** [comment]
please try this chart https://github.com/93Alliance/ng-chartjs

**Comment by miller45** [comment]
@mariusl-a the fix was never "accepted" because no one reviews the pull requests any more. 
Using observables in this case would not solve the problem because the underlying chart has to be redrawn (its html5 canvas) if the labels (not datalabels) in a piechart change.
Besides that 'async to sync' split hack with <ng-container *ngIf="data$ | async; let graph"> is really interesting...I will use it in the future)

**Comment by paviad** [comment]
I am now definitely maintaining and updating this package. Can you provide a stackblitz reproducing this issue?

**Comment by paviad** [comment]
Closed by ef5d2f4
